src/main/java/net/snowflake/client/core/SFBaseSession.java

```diff
@@ -360,27 +360,44 @@ public HttpClientSettingsKey getHttpClientKey() throws SnowflakeSQLException {
       String httpProxyPort = systemGetProperty("http.proxyPort");
       String httpsProxyHost = systemGetProperty("https.proxyHost");
       String httpsProxyPort = systemGetProperty("https.proxyPort");
+      String httpProxyProtocol = systemGetProperty("http.proxyProtocol");
       String noProxy = systemGetEnv("NO_PROXY");
       String nonProxyHosts = systemGetProperty("http.nonProxyHosts");
       // log the JVM parameters that are being used
       if (httpUseProxy) {
         logger.debug(
             "http.useProxy={}, http.proxyHost={}, http.proxyPort={}, https.proxyHost={},"
-                + " https.proxyPort={}, http.nonProxyHosts={}, NO_PROXY={}",
+                + " https.proxyPort={}, http.nonProxyHosts={}, NO_PROXY={}, http.proxyProtocol={}",
             httpUseProxy,
             httpProxyHost,
             httpProxyPort,
             httpsProxyHost,
             httpsProxyPort,
             nonProxyHosts,
-            noProxy);
+            noProxy,
+            httpProxyProtocol);
         // There are 2 possible parameters for non proxy hosts that can be combined into 1
         String combinedNonProxyHosts = Strings.isNullOrEmpty(nonProxyHosts) ? "" : nonProxyHosts;
         if (!Strings.isNullOrEmpty(noProxy)) {
           combinedNonProxyHosts += combinedNonProxyHosts.length() == 0 ? "" : "|";
           combinedNonProxyHosts += noProxy;
         }
-        if (!Strings.isNullOrEmpty(httpsProxyHost) && !Strings.isNullOrEmpty(httpsProxyPort)) {
+
+        // It is possible that a user can have both http and https proxies specified in the JVM
+        // parameters. The default protocol is http.
+        String proxyProtocol = "http";
```

Comment:
Why default to http?

Reply:
To keep it consistent with the default for specifying a proxy server in the connection string settings: https://docs.snowflake.com/en/user-guide/jdbc-parameters#label-jdbc-connection-parameters-proxy-protocol